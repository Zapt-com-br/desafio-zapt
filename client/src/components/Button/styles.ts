import styled, { css } from 'styled-components'
import { darken, lighten } from 'polished'

export const Wrapper = styled.button`
  ${({ theme }) => css`
    padding: ${theme.spacings.xsmall} ${theme.spacings.medium};
    background-color: ${theme.colors.primary};
    color: ${theme.colors.white};
    border: none;

    border-radius: ${theme.border.radius};

    font-size: ${theme.font.sizes.small};

    display: inline-flex;
    text-decoration: none;
    align-items: center;
    text-align: center;
    letter-spacing: -0.01em;

    transition: all 0.3s;

    &:focus {
      box-shadow: 0 0 0 2px ${darken(0.1, theme.colors.secondary)};
    }

    :hover {
      background: ${lighten(0.06, theme.colors.primary)};
    }
  `}
`
