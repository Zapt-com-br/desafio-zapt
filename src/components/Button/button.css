.btn-main {
  min-width: 161px;
  height: 51px;
  cursor: pointer;
  color: #fff;
  border: none;
  border-radius: 4px;
  background-color: var(--blue);
  font-size: 14px;
}