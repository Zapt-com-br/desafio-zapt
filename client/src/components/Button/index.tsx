import { AnchorHTMLAttributes, ButtonHTMLAttributes } from 'react'

import * as S from './styles'

type ButtonTypes =
  | AnchorHTMLAttributes<HTMLAnchorElement>
  | ButtonHTMLAttributes<HTMLButtonElement>

export type ButtonProps = {
  as?: React.ElementType
} & ButtonTypes

const Button = ({ children, as, ...props }: ButtonProps) => (
  <S.Wrapper as={as} {...props}>
    {children}
  </S.Wrapper>
)

export default Button
