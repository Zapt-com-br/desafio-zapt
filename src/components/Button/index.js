import React from 'react';
import './button.css';

function Button({ title }) {
  return (
    <button className="btn-main">{title}</button>
  );
}

export default Button;