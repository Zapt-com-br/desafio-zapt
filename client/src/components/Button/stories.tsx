import { Story, Meta } from '@storybook/react'
import Button, { ButtonProps } from '.'

export default {
  title: 'Button',
  component: Button,
  argTypes: {
    children: {
      type: 'string'
    }
  }
} as Meta

export const Default: Story<ButtonProps> = (args) => <Button {...args} />

Default.args = {
  children: 'Entrar em um grupo'
}

export const asLink: Story<ButtonProps> = ({ ...args }) => <Button {...args} />

asLink.args = {
  children: 'Criar um grupo',
  as: 'a',
  href: '/link'
}
