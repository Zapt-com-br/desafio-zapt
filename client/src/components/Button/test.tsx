import { screen } from '@testing-library/react'
import renderWithTheme from '../../utils/tests/test-util'

import Button from '.'

describe('<Button />', () => {
  it('should render the button correctly', () => {
    const { container } = renderWithTheme(<Button>Entrar em grupo</Button>)

    expect(
      screen.getByRole('button', { name: /Entrar em grupo/i })
    ).toHaveStyle({
      padding: '1.6rem 3.2rem',
      'font-size': '1.4rem'
    })

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render Button as a link', () => {
    renderWithTheme(
      <Button as="a" href="/link">
        Entrar em grupo
      </Button>
    )

    expect(
      screen.getByRole('link', { name: /Entrar em grupo/i })
    ).toHaveAttribute('href', '/link')
  })
})
