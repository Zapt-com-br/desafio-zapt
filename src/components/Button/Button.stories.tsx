import React from 'react';

import { Story, Meta } from '@storybook/react/types-6-0';
import Button, { ButtonProps } from '.';

export default {
  title: 'Generic/Button',
  component: Button,
} as Meta;

export const ButtonDefault: Story<ButtonProps> = args => <Button {...args} />;

ButtonDefault.args = {
  label: 'Criar um grupo',
};
