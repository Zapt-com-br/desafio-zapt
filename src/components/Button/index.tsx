import React, { ButtonHTMLAttributes } from 'react';
import * as Styled from './styles';

import { Typography } from 'components';

export type ButtonProps = {
  label: string;
  onClick?: () => void;
} & ButtonHTMLAttributes<HTMLButtonElement>;

const Button = ({ label, onClick }: ButtonProps) => (
  <Styled.Wrapper onClick={onClick}>
    <Typography color="white" size="small">
      {label}
    </Typography>
  </Styled.Wrapper>
);

export default Button;
