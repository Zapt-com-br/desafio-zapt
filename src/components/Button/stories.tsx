import { Story } from '@storybook/react'
import {
  BoxArrowLeft,
  CheckCircle,
  ExclamationTriangle,
  GearFill,
  InfoCircle,
  PencilSquare,
  Save,
  Trash,
  X,
  XCircle
} from '@styled-icons/bootstrap'
import styled from 'styled-components'
import Button, { BtnProps } from '.'

const WrapperBtn = styled.div`
  display: flex;
  gap: 2rem;
  flex-wrap: wrap;
  justify-content: center;
`

export const Template: Story<BtnProps> = args => (
  <WrapperBtn>
    <Button {...args} />
    <Button {...args} icon={<Save />} />
    <Button {...args} text="" icon={<GearFill />} />
    <Button {...args} text="" icon={<Save />} />
    <Button {...args} text="" icon={<PencilSquare />} />
    <Button {...args} text="" icon={<Trash />} />
    <Button {...args} text="" icon={<X />} />
    <Button {...args} text="" icon={<InfoCircle />} />
    <Button {...args} text="" icon={<ExclamationTriangle />} />
    <Button {...args} text="" icon={<CheckCircle />} />
    <Button {...args} text="" icon={<XCircle />} />
    <Button {...args} text="" icon={<BoxArrowLeft />} />
  </WrapperBtn>
)

export const Default = Template.bind({})
