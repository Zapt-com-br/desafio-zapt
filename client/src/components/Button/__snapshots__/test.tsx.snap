// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the heading 1`] = `
.c0 {
  padding: 1.6rem 3.2rem;
  background-color: #334BC8;
  color: #FFFFFF;
  border: none;
  border-radius: 0.4rem;
  font-size: 1.4rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  -webkit-letter-spacing: -0.01em;
  -moz-letter-spacing: -0.01em;
  -ms-letter-spacing: -0.01em;
  letter-spacing: -0.01em;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c0:focus {
  box-shadow: 0 0 0 2px #ff2925;
}

.c0:hover {
  background: #495fd1;
}

<button
  class="c0"
>
  Entrar em grupo
</button>
`;
